pypesto/hierarchical/inner_calculator_collector.py

```diff
@@ -476,7 +476,10 @@ def __call__(
                 snllh += inner_result[GRAD]
 
             all_inner_pars.update(inner_result[X_INNER_OPT])
-            if INNER_PARAMETERS in inner_result:
+            if (
+                INNER_PARAMETERS in inner_result
+                and inner_result[INNER_PARAMETERS] is not None
+            ):
                 interpretable_inner_pars.extend(inner_result[INNER_PARAMETERS])
```

Comment:
```suggestion
            interpretable_inner_pars.extend(inner_result.get(INNER_PARAMETERS) or [])
```

Reply:
Ohh that's cleaner, ty :D

Reply:
```python
In [1]: {}.get(1, None) or []
Out[1]: []

In [2]: {1:2}.get(1, None) or []
Out[2]: 2
```

Reply:
Okay, if `get` produces a numpy array, this raises a DeprecationWarning. Not ideal then.

Reply:
Hmm something else is breaking as well in the notebook, will find a solution fixing both.

Reply:
fine to keep the original code

Reply:
Made something in between, not a one-liner but clean enough :D